### PR TITLE
added  data folder

### PR DESCRIPTION
**Pull request by roares**

### CODE REVIEW

**Comment by WillemJiang** [comment]
It's better to have a python file to process the data.
We can setup a CI step to call this python script to generate the data.

File: script/spider.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    return dict
+
+
+def getTag(soup):
+
+    try:
+        tag_list = [tag.text.strip() for tag in soup.find_all('a',class_='topic-tag')]
+        tagStr = ','.join(tag_list)
+    except:
+        print(tag_list)
```
Comment (WillemJiang) [inline script/spider.py:58]: Please print the error message instead of tag_list here.

File: script/spider.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    return tagStr
+
+
+def getDetail(soup):
+
+    try: 
+        name = soup.find('a',class_='d-inline-block').text.strip()
+        language = soup.find('span',class_='mr-3').text.strip()
+    except:
+        print(soup.find('span',class_='mr-3'))
```
Comment (WillemJiang) [inline script/spider.py:70]: We need print the error message here.

File: script/stats.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
+import json
+import pandas as pd
+from pandas import NA
+import numpy  as np
+import datetime
+from urllib.request import urlopen
+
+
+# Show all data
```
Comment (WillemJiang) [inline script/stats.py:9]: I don't think we need to do this kind setting here.

File: script/stats.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+
+
+def dataFilter(raw_df):
+    
+    committee_list = raw_df['committee']
+    drop_list = []
+    for committee_name in committee_list:
+        sub_df = raw_df[raw_df['committee'] == committee_name] 
+        if sub_df['size'].max() < 15:
```
Comment (WillemJiang) [inline script/stats.py:21]: Why do you check the committee size here?
Could you add some comments on what kind of work that the dataFilter does?

File: script/stats.py [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    df.rename(columns={'name':'add'},inplace=True)
+    result_dict = completeMonthColumn(df)
+    return result_dict
+    
+
+
+def getCommitter(committees):
+    # Create a datafame, which contains the members' information for each committee: committee name, name, id, date
+    committer = pd.DataFrame()
+    print(committees)
```
Comment (WillemJiang) [inline script/stats.py:130]: I don't think we need the print here.

File: script/stats.py [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+    for item in committees:
+        committee_info = pd.DataFrame.from_dict(committees[item]['roster'],orient='index',columns=['name','date'])
+        committee_info['committee'] = item
+        committee_info['description'] = committees[item]['description']
+
+        committer = pd.concat([committer,committee_info],axis = 0)
```
Comment (WillemJiang) [inline script/stats.py:136]: It's "pmc-member" instead of "committer"

**Review by WillemJiang — CHANGES_REQUESTED** [review]

File: script/stats.py (hunk 4 above)
Comment (roares) [reply script/stats.py]: This is a meaningless function that I forgot to delete.

File: script/spider.py [hunk 7]
```diff
@@ -61,22 +60,54 @@ def getTag(soup):
...
+        "programmingLanguage":programmingLanguage,
+        "issue":issue,
+        "star":star,
+        "pr":pr,
+        "fork":fork
+    }
 
 
-def Merge(dict1, dict2): 
+def Merge(dict1, name_str, dict2):
```
Comment (WillemJiang) [inline script/spider.py:89]: Please add some comments on the Merge method. It's hard to understand the code without comments.

File: src/components/ChartBox.vue [hunk 8]
```diff
@@ -27,7 +27,7 @@ export default {
     methods:{
         getData(){
             const request = new XMLHttpRequest();
-            request.open("get",'/json/committee.json');
+            request.open("get",'/json/committer.json');
```
Comment (WillemJiang) [inline src/components/ChartBox.vue:30]: It should be "comittee" instead of "committer".

**Pull request merged**